lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -725,14 +722,35 @@ mkDelegationCertificates da accXPub =
        Quit -> [toStakeKeyDeregCert accXPub]
 
 
+-- | Describes modifications that can be made to a `Tx` using `updateTx`.
+data TxUpdate = TxUpdate
+    { extraInputs :: [(TxIn, TxOut)]
+    , extraCollateral :: [TxIn]
+       -- ^ Only used in the Alonzo era and later. Will be silently ignored in
+       -- previous eras.
+    , extraOutputs :: [TxOut]
+    , extraInputScripts :: [Script KeyHash]
+    , feeUpdate :: TxFeeUpdate
+        -- ^ Set a new fee or use the old one.
+    }
+
 -- | For testing that
 -- @
---   forall tx. updateSealedTx noTxUpdate tx
+--   forall tx. updateTx noTxUpdate tx
 --      == Right tx or Left
 -- @
 noTxUpdate :: TxUpdate
 noTxUpdate = TxUpdate [] [] [] [] UseOldTxFee
 
+-- | Method to use when updating the fee of a transaction.
+data TxFeeUpdate = UseOldTxFee
+                 -- ^ Instead of updating the fee, just use the old fee of the
+                 -- Tx (no-op for fee update).
+                 | UseNewTxFee Coin
+                 -- ^ Specify a new fee to use instead.
+    deriving (Eq, Show)
```

Comment:
I realise this was just moved from somewhere else, but might as well fix the indentation while we're here:

```suggestion
data TxFeeUpdate
    = UseOldTxFee
        -- ^ Instead of updating the fee, just use the old fee of the
        -- Tx (no-op for fee update).
    | UseNewTxFee Coin
        -- ^ Specify a new fee to use instead.
    deriving (Eq, Show)
```